src/lib/contract.ts

```diff
@@ -49,14 +50,33 @@ export class Contract {
     return readJSON(path.resolve(this.artifactsPath, `${this.moduleName}.json`));
   }
 
+  async getBuildMode(): Promise<BuildMode> {
+    const check = await this.artifactsExist();
+    if (!check.result && check.missingTypes.includes(".contract")) {
+      throw new FileError(
+        `Cannot read .contract bundle, path not found: ${check.missingPaths.join(', ')}`
+      );
+    }
+
+    const contractFilePath = path.resolve(this.artifactsPath, `${this.moduleName}.json`);
```

Comment:
if I'm not mistaken, in case when only `.json` file is missing, we will fail at the 61st line; maybe let's just check only `check.result` (?)

Reply:
yes, it should be enough @prxgr4mm3r